src/tools/kotlin-schema-generator.ts

```diff
@@ -53,7 +54,11 @@ async function visitSchemaFields(schema: Schema, visitor: (field: SchemaField) =
   for (const [field, descriptor] of Object.entries(schema.fields)) {
     switch (descriptor.kind) {
       case 'schema-collection':
-        visitor({field, isCollection: true, schemaType: await getSchemaType(field, descriptor.getFieldType())});
+        visitor({
+          field,
+          isCollection: true,
+          schemaType: await getSchemaType(field, descriptor.getFieldType(), descriptor.annotations)
```

Comment:
how come we don't need to do this for other fields, e.g. ordered-list?
(basically, what if the test case also contained `h: writes Thing {t: List<&Hr> @hardRef}`?)

Reply:
you are right, added to ordered list as well. I don't think there are any other field types to fix?